File: hr-platform/src/main/java/com/example/hrplatform/service/CandidateService.java
package com.example.hrplatform.service;

import com.example.hrplatform.dao.CandidateDAO;
import com.example.hrplatform.model.Candidate;
import com.example.hrplatform.model.Skill;
import java.sql.SQLException;
import java.util.List;

public class CandidateService {
    private final CandidateDAO candidateDAO;

    public CandidateService(CandidateDAO candidateDAO) {
        this.candidateDAO = candidateDAO;
    }

    public Candidate addCandidate(Candidate candidate) throws SQLException {
        return candidateDAO.createCandidate(candidate);
    }

    public Candidate updateCandidate(int id, Candidate candidate) throws SQLException {
        return candidateDAO.updateCandidate(id, candidate);
    }

    public void deleteCandidate(int candidateId) throws SQLException {
        candidateDAO.deleteCandidate(candidateId);
    }

    public Candidate getCandidateById(int candidateId) {
        return candidateDAO.getCandidateById(candidateId);
    }

    public List<Candidate> getAllCandidates() {
        return candidateDAO.getAllCandidates();
    }

    public List<Candidate> searchCandidatesByName(String name) throws SQLException {
        return candidateDAO.searchCandidatesByName(name);
    }

    public List<Candidate> searchCandidatesBySkill(String skill) {
        return candidateDAO.searchCandidatesBySkill(skill);
    }

    public Candidate addSkillToCandidate(int candidateId, int skillId) {
       return candidateDAO.addSkillToCandidate(candidateId, skillId);
    }

    public Candidate removeSkillFromCandidate(int candidateId, int skillId) {
        return candidateDAO.removeSkillFromCandidate(candidateId, skillId);
    }

    public List<Skill> getCandidateSkillsById(int candidateId) {
        return  candidateDAO.getCandidateSkillsById(candidateId);
    }
}

File: front/hr-project-front/src/services/CandidateService.js
import axios from 'axios';
const LOCAL_URL ='http://localhost:8085'
const BASE_URL = LOCAL_URL + '/candidates';


const getAllCandidates = async () => {
  const response = await axios.get(BASE_URL);
  return response.data;
};


const getCandidateById = async (id) => {
  const response = await axios.get(`${BASE_URL}/${id}`);
  return response.data;
};


const addCandidate = async (candidate) => {
  const response = await axios.post(BASE_URL, candidate);
  return response.data;
};

const updateCandidate = async (id, candidate) => {
  const response = await axios.put(`${BASE_URL}/${id}`, candidate);
  return response.data;
};


const deleteCandidate = async (id) => {
  await axios.delete(`${BASE_URL}/${id}`);
};


const searchCandidatesBySkill = async (skill) => {
  const response = await axios.get(`${BASE_URL}/search?skill=${skill}`);
  return response.data;
};

const searchCandidatesBySkill2 = async (skill) => {
  const response = await axios.get(`${BASE_URL}/search2?skill=${skill}`);
  return response.data;
};

const searchCandidatesByName = async (name) => {
  const response = await axios.get(`${BASE_URL}/searchByName?name=${name}`);
  return response.data;
};


const addSkillToCandidate = async (id, skillId) => {
  const response = await axios.post(`${BASE_URL}/${id}/skills?skillId=${skillId}`);
  return response.data;
};


const removeSkillFromCandidate = async (id, skillId) => {
  const response = await axios.delete(`${BASE_URL}/${id}/skills/${skillId}`);
  return response.data;
};

export {
  getAllCandidates,
  getCandidateById,
  addCandidate,
  updateCandidate,
  deleteCandidate,
  searchCandidatesBySkill,
  searchCandidatesBySkill2,
  searchCandidatesByName,
  addSkillToCandidate,
  removeSkillFromCandidate,
};
